flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseTableSourceITCase.java

```diff
@@ -0,0 +1,117 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.addons.hbase.example;
+
+import org.apache.flink.addons.hbase.HBaseTableSource;
+import org.apache.flink.addons.hbase.HBaseTestingClusterAutostarter;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableConfig;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.java.BatchTableEnvironment;
+import org.apache.flink.table.sources.BatchTableSource;
+import org.apache.flink.types.Row;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.Test;
+import scala.tools.cmd.gen.AnyVals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class HBaseTableSourceITCase extends HBaseTestingClusterAutostarter {
+
+
+	public static final byte[] ROW_1 = Bytes.toBytes("row1");
+	public static final byte[] ROW_2 = Bytes.toBytes("row2");
+	public static final byte[] ROW_3 = Bytes.toBytes("row3");
+	public static final byte[] F_1 = Bytes.toBytes("f1");
+	public static final byte[] Q_1 = Bytes.toBytes("q1");
+	public static final byte[] Q_2 = Bytes.toBytes("q2");
+	public static final byte[] Q_3 = Bytes.toBytes("q3");
+
+	@Test
+	public void testHBaseTableSource() throws Exception {
+		// create a table with single region
+		TableName tableName = TableName.valueOf("test");
+		createTable(tableName, F_1, new byte[1][]);
+		// get the htable instance
+		HTable table = openTable(tableName);
+		List<Put> puts = new ArrayList<Put>();
+		// add some data
+		Put put = new Put(ROW_1);
+		// add 3 qualifiers per row
+		//1st qual is integer
+		put.addColumn(F_1, Q_1, Bytes.toBytes(100));
+		//2nd qual is String
+		put.addColumn(F_1, Q_2, Bytes.toBytes("strvalue"));
+		// 3rd qual is long
+		put.addColumn(F_1, Q_2, Bytes.toBytes(19991l));
+		puts.add(put);
+
+		put = new Put(ROW_2);
+		// add 3 qualifiers per row
+		//1st qual is integer
+		put.addColumn(F_1, Q_1, Bytes.toBytes(101));
+		//2nd qual is String
+		put.addColumn(F_1, Q_2, Bytes.toBytes("strvalue1"));
+		// 3rd qual is long
+		put.addColumn(F_1, Q_2, Bytes.toBytes(19992l));
+		puts.add(put);
+
+		put = new Put(ROW_3);
+		// add 3 qualifiers per row
+		//1st qual is integer
+		put.addColumn(F_1, Q_1, Bytes.toBytes(102));
+		//2nd qual is String
+		put.addColumn(F_1, Q_2, Bytes.toBytes("strvalue2"));
+		// 3rd qual is long
+		put.addColumn(F_1, Q_2, Bytes.toBytes(19993l));
```

Comment:
Ya.  Good catch. Will change it. I was more focussed on the impl because if you see the test case itself was not asserting any. I need to add all those stuff.